### ISSUE TITLE
Video colorizer downscales 4K videos to 1080p videos

### ISSUE THREAD

**Issue by IveMalfunctioned** [jantic/DeOldify#425]

**Comment by jantic** [comment]
This isn't actually true strictly speaking- 4k will output as 4k in terms of dimensions.  **But**:

- It does convert frames to jpg though which may be causing a small visible degradation however- an engineering trade-off for practical space usage and wide ability to deploy.
-  Colors are rendered at much lower resolutions than 4k- default is at 336x336  (render_factor 21) and it  goes no higher than about 640x640 (render_factor 40). This can definitely cause visible degradation in the case of 4k, even after taking the effect of human perception of color into account. Chroma-subsampling has its limits here, after all.  This really can't be practically changed unfortunately without a brand new model (I've done it but only for commercial purposes). Basically that change isn't going to happen soon.

So basically what I'm saying is- Yes, there's limitations to the renders which you are probably seeing here but I don't have a solution for you. Sorry about that.
